src/components/organisms/RegisterForm/RegisterForm.tsx

```diff
@@ -124,6 +131,28 @@ export const RegisterForm: React.FC<{
     [fields]
   );
 
+  const getFieldTranslate = useCallback(() => {
+    const currLang: string = i18n.language;
+
+    fields.list &&
+      fields.list.map((field) => {
+        if (Array.isArray(field.extra)) {
+          const translationFromServer = field.extra.find(
+            (item) => currLang in item
+          );
+          if (translationFromServer) {
+            i18n.addResourceBundle(currLang, "translation", {
+              [`RegisterForm.${field.name}`]: translationFromServer[currLang],
+            });
+          }
+        }
+      });
+  }, [fields, i18n]);
+
+  useEffect(() => {
+    getFieldTranslate();
+  }, [fields]);
```

Comment:
this is not part of this component .... 

remove it from here, you can add this later when implementing page or elsewhere